src/pages/MealPage/MealPage.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { FieldValues, SubmitHandler, useForm, FormProvider } from 'react-hook-form';
+import { useRestaurants } from '../../utils/hooks/useRestaurants/useRestaurants';
+import { useBasket } from '../../utils/hooks/useBasket/useBasket';
+import MealPopup from './MealPopup/MealPopup';
+import MealImage from './MealImage/MealImage';
+import MealDescription from './MealDescription/MealDescription';
+import MealTotal from './MealTotal/MealTotal';
+import MealFeatureList from './MealFeatureList/MealFeatureList';
+import Preloader from '../../components/Preloader/Preloader';
+import { Feature, Meal, Restaurant } from '../../utils/api/restaurantsService/restaurantsService';
+import { sumBy } from 'lodash';
+
+function MealPage() {
+    const [features, setFeatures] = useState<Feature[]>([]);
+    const navigate = useNavigate();
+    const params = useParams();
+    const { restaurantsFiltered } = useRestaurants();
+    const { addMeal, isLoading } = useBasket();
+    const methods = useForm();
+    const { watch } = methods;
+    const restaurant: Restaurant | undefined = restaurantsFiltered.find((restaurant) => restaurant.id === params.restaurantId);
+    const meal: Meal | undefined = restaurant && restaurant.meals.find((meal) => meal.id === params.mealId);
+    const price = sumBy(features, (feature) => feature.choices.filter((choice) => choice.default)[0].price);
+    const goBack = () => {
+        navigate(`/restaurants/${params.restaurantId}`);
+    };
+    const close = () => {
+        navigate('/restaurants');
+    };
+    useEffect(() => {
+        const formValues = watch((value, { name }) => {
+            const nextFeatures = features.map((feature: Feature) => {
+                if (feature.name === name) {
+                    const choices = feature.choices.map((choice) => {
+                        if (choice.name === value[name]) {
+                            return { ...choice, default: true };
+                        } else return { ...choice, default: false };
```

Comment:
```suggestion
                        return { ...choice, default: choice.name === value[name]};
```

Когда происходят какие-то вычисления, и результатом их явно пишется return true или false, то я сразу подозреваю, именно строку с true или false можно опустить, а оставить только последнее вычисление

Reply:
починил